catalog/resource_provider.go

```diff
@@ -51,15 +57,19 @@ func (a ProvidersAPI) updateProvider(ci *ProviderInfo) error {
 }
 
 func ResourceProvider() *schema.Resource {
-	providerSchema := common.StructToSchema(ProviderInfo{}, func(m map[string]*schema.Schema) map[string]*schema.Schema {
```

Comment:
here's how we can make this more readable and clear:

1. revert the `providerSchema` rename. It diverts the attention from the intention of this commit by making more changes than necessary.
2. remove `ProviderInfoMetadata` as it's not necessary anymore.
3. create the copy of `providerSchema`, but without `recipient_profile_str`:

```go
providerSchemaForRead := map[string]*schema.Schema{
  "name": providerSchema["name"],
  "comment": providerSchema["comment"],
  "authentication_type": providerSchema["authentication_type"],
}
```

you can also just inline that in the `Read` callback.

Reply:
Done